.github/workflows/ignore-e2e-fork.yml

```diff
@@ -0,0 +1,21 @@
+# This workflow is added in order for the required e2e/build job to run successfully even
+# when the PR targets non-code files (e.g. docs, LICENSE, etc). This is done because of the
+# way Github required checks work, reference:
+#
+# https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/troubleshooting-required-status-checks#handling-skipped-but-required-checks
+name: Tests / E2E Fork
+on:
```

Comment:
should we run this on non-fork branches as well?

Reply:
I think this passes fine for tests whose workflow jobs are skipped with an `if` (i.e, the result of the job is Skipped and not Pending)

Reply:
Seems Github makes a distinction on skipping the whole workflow and skipping all the jobs of a workflow.

Skipping the workflow as a whole is done with path filtering, branch filtering as mentioned in the [docs](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/troubleshooting-required-status-checks#handling-skipped-but-required-checks). Doing this skip results in its status being reported as 'Pending' thereby blocking the required checks from being fulfilled.

Skipping jobs inside a workflow file can be done using expressions, we currently do this to not run our e2es on drafts e.g [here](https://github.com/cosmos/ibc-go/blob/b5dadfdd99addb381fbf0b8ca0e7070d5aff7891/.github/workflows/e2e.yaml#L21). Doing these skips marks the job as being 'Success', allowing the required checks to complete.

>  Just want to double check that if we mark e2e fork tests as required, if I open a pr against main which makes code changes, that test will be skipped for me

If my understanding of their docs is correct, if we mark e2e-fork as required, this shouldn't hold up a pull request.

Reply:
That seems to make sense to me, but I'm just confused why marking both fork and non-fork e2e tests previously resulted in a "pending" state for pr's? I tried this with fork pr's by marking an e2e test as required and then updating a dependabot pr which should the test in "pending"